mockgen/import_mode.go

```diff
@@ -0,0 +1,376 @@
+package main
+
+import (
+	"fmt"
+	"go/token"
+	"go/types"
+	"math/rand"
+	"os"
+
+	"go.uber.org/mock/mockgen/model"
+	"golang.org/x/tools/go/packages"
+)
+
+type importModeParser struct {
+	pkgName string
+}
+
+func (p *importModeParser) parsePackage(packageName string, ifaces []string) (*model.Package, error) {
+	p.pkgName = packageName
+
+	pkg, err := p.loadPackage(packageName)
+	if err != nil {
+		return nil, fmt.Errorf("load package: %w", err)
+	}
+
+	interfaces, err := p.extractInterfacesFromPackage(pkg, ifaces)
+	if err != nil {
+		return nil, fmt.Errorf("extract interfaces from package: %w", err)
+	}
+
+	return &model.Package{
+		Name:       pkg.Types.Name(),
+		PkgPath:    packageName,
+		Interfaces: interfaces,
+	}, nil
+}
+
+func (p *importModeParser) loadPackage(packageName string) (*packages.Package, error) {
+	const fakeFileTemplate = `
+		package main
+		
+		import (
+			"%s"
+		)
+	`
+
+	dirName := "mockgen_" + p.generateSalt()
+	err := os.Mkdir(dirName, 0755)
+	if err != nil {
+		return nil, fmt.Errorf("create temp dir for fake package: %w", err)
+	}
+	defer os.RemoveAll(dirName)
+
+	err = os.WriteFile(dirName+"/main.go", []byte(fmt.Sprintf(fakeFileTemplate, packageName)), 0666)
+	if err != nil {
+		return nil, fmt.Errorf("write fake main.go: %w", err)
+	}
+
+	fileSet := token.NewFileSet()
+
+	cfg := &packages.Config{
+		Mode: packages.NeedDeps | packages.NeedImports | packages.NeedTypes | packages.NeedTypesInfo | packages.NeedEmbedFiles,
+		Dir:  dirName,
+		Fset: fileSet,
+	}
+	pkgs, err := packages.Load(cfg)
+	if err != nil {
+		return nil, fmt.Errorf("load packages: %w", err)
+	}
+
+	if len(pkgs) != 1 {
+		return nil, fmt.Errorf("packages length must be 1: %d", len(pkgs))
+	}
+
+	pkg := pkgs[0]
+	searchedPkg, ok := pkg.Imports[packageName]
+	if !ok {
+		return nil, fmt.Errorf("package %s not found", packageName)
+	}
+
+	return searchedPkg, nil
+}
+
+func (p *importModeParser) generateSalt() string {
+	var numbers = []byte("0123456789")
+
+	result := make([]byte, 8)
+	for i := range result {
+		result[i] = numbers[rand.Intn(len(numbers))]
+	}
+
+	return string(result)
+}
+
+func (p *importModeParser) extractInterfacesFromPackage(pkg *packages.Package, ifaces []string) ([]*model.Interface, error) {
+	interfaces := make([]*model.Interface, len(ifaces))
+	for i, iface := range ifaces {
+		obj := pkg.Types.Scope().Lookup(iface)
+		if obj == nil {
+			return nil, fmt.Errorf("interface %s does not exist", iface)
+		}
+
+		modelIface, err := p.parseInterface(obj)
+		if err != nil {
+			return nil, newParseTypeError("parse interface", obj.String(), err)
+		}
+
+		interfaces[i] = modelIface
+	}
+
+	return interfaces, nil
+}
+
+func (p *importModeParser) parseInterface(obj types.Object) (*model.Interface, error) {
+	named, ok := obj.Type().(*types.Named)
```

Comment:
Here is where we'll want to probably use `types.Unalias` to support aliases in go 1.23.
```suggestion
	named, ok := types.Unalias(obj.Type()).(*types.Named)
```